test/int/project_test.go

```diff
@@ -388,12 +389,13 @@ var _ = Describe("AtlasProject", Label("int", "AtlasProject"), func() {
 				connectionSecret.StringData["publicApiKey"] = "non-existing"
 				Expect(k8sClient.Update(context.Background(), &connectionSecret)).To(Succeed())
 
-				Expect(k8sClient.Get(context.Background(), kube.ObjectKeyFromObject(createdProject), createdProject)).To(Succeed())
-				createdProject.Spec.AlertConfigurationSyncEnabled = true
-				Expect(k8sClient.Update(context.Background(), createdProject)).To(Succeed())
-				Expect(k8sClient.Get(context.Background(), kube.ObjectKeyFromObject(secondProject), secondProject)).To(Succeed())
-				secondProject.Spec.AlertConfigurationSyncEnabled = true
-				Expect(k8sClient.Update(context.Background(), secondProject)).To(Succeed())
+				Eventually(updateK8sObject[*mdbv1.AtlasProject](createdProject, func(createdProject *mdbv1.AtlasProject) {
+					createdProject.Spec.AlertConfigurationSyncEnabled = true
+				})).WithPolling(interval).WithTimeout(20 * time.Second).Should(BeTrue())
+
+				Eventually(updateK8sObject[*mdbv1.AtlasProject](secondProject, func(secondProject *mdbv1.AtlasProject) {
+					secondProject.Spec.AlertConfigurationSyncEnabled = true
+				})).WithPolling(interval).WithTimeout(20 * time.Second).Should(BeTrue())
```

Comment:
@josvazg  Looks good in general, but is there a reason to extract the logic into a function that is only called once? Consider moving the body of the function here using `Gomega`. That is way simpler than adding a generic function that takes another function